### ISSUE TITLE
.NET 5.0 SDK breaks old projects

### ISSUE THREAD

**Issue by richlander** [dotnet/sdk#10812]
I had a working 3.1 environment and then installed .NET 5.0 Preview 1, which then broke my environment based on its (apparent) understanding of the minimum .NET Core 3.1 runtime version.

```console
rich@thundera fooapp % cat fooapp.csproj 
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

</Project>
rich@thundera fooapp % dotnet --version
5.0.100-preview.1.20155.7
rich@thundera fooapp % dotnet run
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '3.1.2' was not found.
  - The following frameworks were found:
      2.1.12 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.0.0-preview7-27912-14 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.0.0-rc1-19426-11 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.0.0 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.1.0-preview1.19506.1 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.1.0-preview2.19525.6 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      3.1.0 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
      5.0.0-preview.1.20120.5 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.2&arch=x64&rid=osx.10.15-x64
rich@thundera fooapp % dotnet new globaljson
The template "global.json file" was created successfully.
rich@thundera fooapp % cat global.json
{
  "sdk": {
    "version": "3.1.100"
  }
}
rich@thundera fooapp % dotnet run
Hello World!
```

**Comment by dsplaisted** [comment]
@sfoslund We need to update this line (and similar for other 3.1 KnownFrameworkReference items): https://github.com/dotnet/core-sdk/blob/edbdf82041abff369ed64e5d0d6fd590e75b5328/src/redist/targets/GenerateBundledVersions.targets#L249

Compare with 3.0: https://github.com/dotnet/core-sdk/blob/edbdf82041abff369ed64e5d0d6fd590e75b5328/src/redist/targets/GenerateBundledVersions.targets#L319

**Comment by richlander** [comment]
Yes. That seems to be the problem.

Won't this line (and others like it) have a similar problem? https://github.com/dotnet/core-sdk/blob/edbdf82041abff369ed64e5d0d6fd590e75b5328/src/redist/targets/GenerateBundledVersions.targets#L268

**Comment by sfoslund** [comment]
Fixed by dotnet/core-sdk#6753
